dataflow_test.py

```diff
@@ -541,6 +543,70 @@ def test_count_filter():
     assert result.n_failed == len(the_source) - len(expected_result)
 
 
+# In dataflow, the source can also generate more complex
+# data structures. Hence, the same entry can gather
+# different types of data in a single object.
+# A useful manner of organizing the data is using some
+# kind of namespace that labels each node of information.
+# In order to work more comfortably in this scenario,
+# most of the basic components in dataflow take optional
+# arguments that allow the user to specify which node of
+# information the component should use. The output of
+# the component can be put back under the same or a
+# different label.
```

Comment:
The motivation behind the namespaces-in-the-stream was that it was the simplest way of enabling parallel, synchronized and joinable streams. (It's a bit of a hack, but it's a damn simple, effective, flexible and easy to understand hack. I find it very unsatisfying from the purity perspective: we're *mutating* (ugh!) a namespace; they're not *really* flow networks/graphs ... but it's very pragmatic.)

Anyway, the documentation of this feature should highlight this parallel, synchronized, joinable flows perspective.

This will require careful thought, and a whole bunch of non-trivial examples that emphasize this aspect.